packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts

```diff
@@ -0,0 +1,146 @@
+      };
+    },
+  ),
+]);
+
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_createWithSystemAssigned =
+  passOnSuccess([
+    mockapi.put(
+      "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+      (req) => {
+        req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+        if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+          throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+        }
+        if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+          throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+        }
+        if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+          throw new ValidationError(
+            "Unexpected managed identity resource name",
+            "identity",
+            req.params.managedIdentityResourceName,
+          );
+        }
+        req.expect.deepEqual(req.body["identity"], expectedIdentity);
+        return {
+          status: 200,
+          body: json(validSystemAssignedManagedIdentityResource),
+        };
+      },
+    ),
+  ]);
+
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_updateWithUserAssignedAndSystemAssigned =
+  passOnSuccess([
+    mockapi.patch(
+      "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+      (req) => {
+        req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+        if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+          throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+        }
+        if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+          throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+        }
+        if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+          throw new ValidationError(
+            "Unexpected managed identity resource name",
+            "identity",
+            req.params.managedIdentityResourceName,
+          );
+        }
+        req.expect.bodyEquals({
+          identity: {
+            type: `${IDENTITY_TYPE_SYSTEM_USER_ASSIGNED_EXPECTED}`,
+            userAssignedIdentities: {
+              "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1":
+                {},
+            },
+          },
+        });
```

Comment:
Same here? Only check "identity".

Reply:
The validation of request body has been fixed to `req.expect.deepEqual(actualBody, expectedBody)`;